Privacy policy and signup mobile

marcusjensen15 opened the pull request:
### What issue is this solving?
This another mobile view for issue #244 -> Privacy Policy and Signup.

### Any helpful knowledge/context for the reviewer?

Note: This branch is a branch off of my previous pull request. PR #291 . If that pull request doesn't get accepted, don't accept this one. 

This branch looks identical to the wireframes, but if you find something incorrect or out of wack, please let me know 👍

marcusjensen15 committed:
Added Google Button Component

marcusjensen15 committed:
create unique signup.scss file and add it to all.scss

marcusjensen15 committed:
change all login classes to signup classes

marcusjensen15 committed:
sucessfully showing/hiding headers based on screensize

marcusjensen15 committed:
spacing for the create an account mobile heading looking correct

marcusjensen15 committed:
sign up button now has correct coloring

marcusjensen15 committed:
blue header spacing looking good. still need to get button sizing correct

marcusjensen15 committed:
style dwellingly logo in top header

marcusjensen15 committed:
add white dwellingly logo to top header

marcusjensen15 committed:
correctly sized signup with google button

marcusjensen15 committed:
remove padding for internal google button text

marcusjensen15 committed:
error messagesd now appearing within the form input cells

marcusjensen15 committed:
make seperate media query for iphone 5

marcusjensen15 committed:
finalize iphone 5 style

marcusjensen15 committed:
spacing adjustments for all mobile sizes

marcusjensen15 committed:
signup views now looking good across all mobile devices. requesting feedback

marcusjensen15 committed:
add link to within signup.js. import link into signup. now routing to home screen because privacy policy doesn't exist yet

marcusjensen15 committed:
built out frame for Privacy Policy view. Exported to app.js, and added route. correctly routing to minimal privacy policy component

marcusjensen15 committed:
mobile header now looking good

marcusjensen15 committed:
correctly nested divs

marcusjensen15 committed:
list styling now looking good

marcusjensen15 committed:
filled in privacy policy with placeholder text. everything looking good

marcusjensen15 committed:
adding the scss file

Chris-Boe commented on src/scss/pages/_privacyPolicy.scss:
```diff
@@ -0,0 +1,71 @@
+// Privacy Policy Styles
+
+
+// .privacy-policy {
```
Think we can get rid of this commented out section and a lot of the empty lines in this file

Chris-Boe commented on src/views/privacyPolicy.js:
```diff
@@ -0,0 +1,84 @@
+              Artisan irony art party fixie vinyl, gochujang mustache yr succulents unicorn letterpress meggings craft beer kinfolk. 
+              Iceland everyday carry farm-to-table blue bottle direct trade messenger bag. 
+        </p>
+
```
There's lots of empty lines I think we can ditch

Chris-Boe commented on src/views/privacyPolicy.js:
```diff
@@ -0,0 +1,84 @@
+
+
+
+const PrivacyPolicy = () => {
```
No need for a change here, but JS does have short hand for an "implicit return" for arrow functions where instead of opening a method body, you just open parentheses for the body and it acts as a "return":

`const PrivacyPolicy = () => (
  div... /div
)`

Chris-Boe commented on src/views/privacyPolicy.js:
```diff
@@ -0,0 +1,84 @@
+    <div className="privacy-policy__text-container">
+      <h3 className="privacy-policy__heading">Privacy Policy</h3>
+        <p className="privacy-policy__text"> 
+              I'm baby blog taxidermy roof party activated charcoal man bun. 
```
"Iceland erryday" 😆 so let's check if we have real text for this, I see that we have some in the mockup but I'm not sure if that's just generic privacy policy stuff

Chris-Boe reviewed (changes requested):
Let's keep playing around with a more _responsive_ view, since this is what I see on screens between 500px - 1200px for the signup page:
![image](https://user-images.githubusercontent.com/8600792/92200271-457dcc80-ee2e-11ea-8530-99d427914b96.png)
and for the privacy policy:
![image](https://user-images.githubusercontent.com/8600792/92200285-52022500-ee2e-11ea-83a2-d6ca1664985d.png)


The existing header and main container in the app are both mobile friendly if we wanted to use them!

Chris-Boe commented:
Also I just realized this was branched off your mobile signup page branch, please make sure you branch off development for new branches! This helps keep the work for issues contained, and if the parent branch gets updated but the sub-branches don't, it can cause some issues

Chris-Boe commented:
Because the header and main-container are responsive, we can actually just use the exact same layout as the /terms page! Feel free to pilfer those styles
![image](https://user-images.githubusercontent.com/8600792/92200690-62ff6600-ee2f-11ea-9cf6-671e0a25511c.png)

marcusjensen15 commented:
Totally, I'll fix this. Quick note on the privacy policy on larger screens, I actually hadn't designed it for that. This PR was just focused on mobile as there were no wireframes for larger screens. I'll get something looking good for the larger screens as well!

marcusjensen15 committed:
revise breakpoint mobile width so design better accomidates mid sized screens

marcusjensen15 committed:
revise privacy policy link styling on normal sized screens

marcusjensen15 committed:
design and implement privacy policy view for desktop sized screens

marcusjensen15 committed:
eleminate unnecessary comments and space

marcusjensen15 commented:
Hey Chris!

I got this looking good from mobile screen widths all the way up to large monitors, it should now be super responsive. Also, I eliminated all of the unnecessary spaces/comments in the scss files.

I left in the placeholder text for privacy policy because I wasn't sure if the policy outlined in the wireframe was the actual privacy policy for dwellingly, or if they just used place holder text as well. I didn't want to spend the time copying it over if it isn't the real policy, and i couldn't find a .txt file containing the real policy. I'm happy to swap it out once we have the real text!

Iceland Erryday haha

marcusjensen15 committed:
Merge branch 'development' into privacy-policy-mobile

Chris-Boe commented on src/scss/pages/_all.scss:
```diff
@@ -13,4 +13,6 @@
-@import './settings';
+@import './signup';
+@import './privacyPolicy';
+
```
So it looks like the `settings` scss file reference got removed here during the merge conflict resolution, so let's add that back

marcusjensen15 replied:
Added back and good to go!

Chris-Boe reviewed (changes requested):
Nice job! Both the privacy policy and signup pages look good and are very responsive!
Just gotta fix one deleted line and it's good to go!

marcusjensen15 committed:
Update _all.scss

Add ./settings into _all.scss. This was removed in my branch because I'm a bit behind development. It didn't exist when i branched off.

marcusjensen15 committed:
Merge branch 'development' into privacy-policy-mobile

marcusjensen15 commented:
Hey Chris! just added that missing import statement and resolved some pretty big conflicts on signup.js (not shocking as i had to re-name a bunch of classes).
